src/extension.ts

```diff
@@ -1,398 +1,194 @@
-            logs.info("Git Extension - Add branch.");
-            context.subscriptions.push(await getBranchListener(context, logs));
-        } else {
-            logs.warn("Git Extension - Could not find active git extension in workspace.");
-        }
-    } else {
-        logs.warn("Git extension - Could not find vscode.git installed.");
-    }
-
-    // SCA Auto Scanning view
-    const scaResultsProvider = new SCAResultsProvider(
-        logs,
-        statusBarItem,
-        diagnosticCollection,
-    );
-    scaResultsProvider.scaResults = [];
-    vscode.window.registerTreeDataProvider(`scaAutoScan`, scaResultsProvider);
-    const scaTree = vscode.window.createTreeView("scaAutoScan", { treeDataProvider: scaResultsProvider });
-    scaTree.onDidChangeSelection((item) => {
-        if (item.selection.length > 0) {
-            if (!item.selection[0].contextValue && !item.selection[0].children) {
-                // Open new details
-                vscode.commands.executeCommand(
-                    "ast-results.newDetails",
-                    item.selection[0].result,
-                    "realtime"
-                );
-            }
-        }
-    });
-    // Settings
-    context.subscriptions.push(vscode.commands.registerCommand(`${EXTENSION_NAME}.viewSettings`, () => {
-        vscode.commands.executeCommand(
-            "workbench.action.openSettings",
-            `@ext:checkmarx.ast-results`
-        );
+  // Create logs channel and make it visible
+  const output = vscode.window.createOutputChannel(constants.extensionFullName);
+  const logs = new Logs(output);
+  logs.show();
+  logs.info(messages.pluginRunning);
+
+  // Status bars for scans from IDE and SCA auto scanning
+  const runScanStatusBar = vscode.window.createStatusBarItem(
+    vscode.StatusBarAlignment.Left
+  );
+  const runSCAScanStatusBar = vscode.window.createStatusBarItem(
+    vscode.StatusBarAlignment.Left
+  );
+  runSCAScanStatusBar.text = messages.scaStatusBarConnect;
+  runSCAScanStatusBar.show();
+
+  await setScanButtonDefaultIfScanIsNotRunning(context);
+
+  // Scans from IDE scanning commands
+  const ideScanCommand = new IDECommand(context, runScanStatusBar, logs);
+  ideScanCommand.registerIdeScans();
+  vscode.commands.executeCommand(commands.pollScan);
+
+  const kicsDiagnosticCollection =
+    vscode.languages.createDiagnosticCollection(constants.extensionName);
+  const kicsStatusBarItem = vscode.window.createStatusBarItem(
+    vscode.StatusBarAlignment.Left
+  );
+  const kicsProvider = new KicsProvider(
+    context,
+    logs,
+    kicsStatusBarItem,
+    kicsDiagnosticCollection,
+    [],
+    []
+  );
+  const kicsScanCommand = new KICSRealtimeCommand(context, kicsProvider, logs);
+  kicsScanCommand.registerKicsScans();
+
+  const diagnosticCollection =
+    vscode.languages.createDiagnosticCollection(constants.extensionName);
+  const statusBarItem = vscode.window.createStatusBarItem(
```

Comment:
@tiagobcx  let's create the status bar all together.

Reply:
Ok, move them all together